### PR TITLE
Update Haskell package set to Stackage Nightly 2021-04-23 (plus other fixes)

### PR DESCRIPTION
**Pull request by peti**
This PR is test-built by Hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. I'll fix up the remaining errors and merge it on **Friday, 2021-04-30 20:00 UTC+2**. Everyone can participate in that process live on Twitch at https://www.twitch.tv/peti343. In addition to the chat features offered by Twitch, there is also a voice conference at https://discord.gg/YTEa3XR viewers can use to chat with me and with each other. 

Please note that this will be **last** merge of `haskell-updates` that I'll do for the foreseeable future. Consequently, this is going to be the last livestream about Haskell/Nix for the foreseeable future, too.

TODO
--------

- [x] Respond to the questions in https://github.com/NixOS/nixpkgs/issues/121140.

### CODE REVIEW

**Comment by peti** [comment]
> https://discourse.nixos.org/t/im-retiring-from-haskell-maintenance-in-nixpkgs/12739/1

tl;dr: The upcoming livestream on Friday is going to be the last one. I'm retiring. Let those young folk maintain Haskell packages in Nix from now. :-)
